### ISSUE TITLE
Why doesn't UUID version 5 give expected results?

### ISSUE THREAD

**Issue by homanchou** [zyro/elixir-uuid#26]
I'm under the impression that UUID version 5, if given the same parameters, on different machines or different implementations should give back the same uuid.

Here is the ossp-uuid linux utility running on my mac terminal:
```
$ uuid -v5 ns:OID "hi.bye"
e1cd0925-0852-5171-9e44-5fce3d17ee7d
```

Here is the uuidutils gem running in ruby irb, as you can see it matches the linux utility.
```
irb> require 'uuidtools'
irb> UUIDTools::UUID.sha1_create(UUIDTools::UUID_OID_NAMESPACE, "hi.bye").to_s
 => "e1cd0925-0852-5171-9e44-5fce3d17ee7d"
```

But here is the output using iex:
```
iex> UUID.uuid5 :oid, "hi.bye"
"cb6215ac-0b53-5366-aa60-52194d5c46da"
```

I expect the elixir implementation to produce the same results as the linux and ruby implementation.

**Comment by homanchou** [comment]
As a work around for now, specifying the uuid string of the namespace does in fact work:
```
> UUID.uuid5("6ba7b812-9dad-11d1-80b4-00c04fd430c8", "hi.bye")
"e1cd0925-0852-5171-9e44-5fce3d17ee7d"
```
Here is a list of uid namespaces:

http://stackoverflow.com/questions/10867405/generating-v5-uuid-what-is-name-and-namespace
```
The UUID RFC pre-defines four namespaces for you:

NameSpace_DNS: {6ba7b810-9dad-11d1-80b4-00c04fd430c8}
NameSpace_URL: {6ba7b811-9dad-11d1-80b4-00c04fd430c8}
NameSpace_OID: {6ba7b812-9dad-11d1-80b4-00c04fd430c8}
NameSpace_X500:{6ba7b814-9dad-11d1-80b4-00c04fd430c8}
```

Using the atoms does not work, it looks like the issue might be in this line
https://github.com/zyro/elixir-uuid/blob/master/lib/uuid.ex#L330

Might be the wrong type of encoding compared to erlang version?  https://github.com/avtobiff/erlang-uuid/blob/master/src/uuid.erl#L193

The nil namespace works, since it's all zeros.  https://github.com/zyro/elixir-uuid/blob/master/lib/uuid.ex#L337

**Comment by zyro** [comment]
You're right, looks like it was a binary encoding issue. Thanks for the thorough investigation!

I've released an updated version: https://hex.pm/packages/uuid/1.1.6

Closed by #27.

**Comment by homanchou** [comment]
cool!  Thanks for merging.  I assume you're going to update the readme too?  I think the uuid3 and uuid5 examples on it might be wrong too.
